### PR TITLE
Add Node and Python unit test workflows

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
### Description:
Add Node and Python unit test workflows

### Testing Steps:
Checks pass

### Please Review
@Workiva/xt @paulwarren-wk

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/node-tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          - macos-12
+          - ubuntu-22.04
+          - windows-2022
+        node-version:
+          - '14'
+          - '15'
+          - '16'
+          - '17'
+          - '18'
+          - '19'
```
Comment (austinmatherne-wk) [inline .github/workflows/node-tests.yml:22]: We can probably drop v15 and v17 as those aren't LTS releases and are already EOL

**Review by sagesmith-wf — DISMISSED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Comment by aaroncameron-wk** [comment]
Looks like we may need actions permissions updated to allow workflows to run on PR. These new workflows are passing on my fork [here](https://github.com/aaroncameron-wk/ixbrl-viewer/pull/12).

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by allenbross-wf** [comment]
Checks passed.
QA +10

@Workiva/release-management-pp

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**